src/main/java/wanion/avaritiaddons/block/chest/infinity/TileEntityInfinityChest.java

```diff
@@ -66,11 +66,16 @@ public final void setInventorySlotContents(int slot, final ItemStack itemStack)
 		markDirty();
 	}
 
+	private boolean isSameItem(ItemStack slotStack, ItemStack itemStack) {
+		if (slotStack == null || itemStack == null) return false;
+		return slotStack.getItem() == itemStack.getItem() && (!itemStack.getHasSubtypes() || itemStack.getItemDamage() == slotStack.getItemDamage()) && ItemStack.areItemStackTagsEqual(itemStack, slotStack);
```

Comment:
what do you think about splitting this condition into parts and giving them descriptive names, it's a bit hard to read, like
```java
boolean itemsEqual = slotStack.getItem() == itemStack.getItem();
....
return itemsEqual && ...;
```